### PR TITLE
Add documentation for `<Kokkos_BitManipulation.hpp>` header

### PR DESCRIPTION
**Pull request by dalg24**
Corresponds to kokkos/kokkos#4577

### CODE REVIEW

File: docs/source/API/core/numerics/bit-manipulation.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+.. |StandarLibraryHeaderBit| replace:: ``<bit>``
+
+Defined in header |KokkosBitManipulation|_ which is included from ``<Kokkos_Core.hpp>``
+
+Provides function templates from the standard library header |StandarLibraryHeaderBit|_ (since C++20).
+
+Bit manipulation function templates are defined in the ``Kokkos::`` namespace since Kokkos 4.1
+
+.. _bit_cast: https://en.cppreference.com/w/cpp/numeric/bit_cast
```
Comment (masterleinad) [inline docs/source/API/core/numerics/bit-manipulation.rst:24]: This doesn't appear as a link.

File: docs/source/API/core/numerics/bit-manipulation.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+.. |bit_cast| replace:: ``bit_cast``
+
+.. _byteswap: https://en.cppreference.com/w/cpp/numeric/byteswap
```
Comment (masterleinad) [inline docs/source/API/core/numerics/bit-manipulation.rst:28]: This doesn't appear as a link. This is C++23.

File: docs/source/API/core/numerics/bit-manipulation.rst [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+.. _bit_floor: https://en.cppreference.com/w/cpp/numeric/bit_floor
+
+.. |bit_floor| replace:: ``bit_floor``
+
+.. _bit_width: https://en.cppreference.com/w/cpp/numeric/bit_width
+
+.. |bit_width| replace:: ``bit_width``
+
+.. _rotl: https://en.cppreference.com/w/cpp/numeric/rotl
```
Comment (masterleinad) [inline docs/source/API/core/numerics/bit-manipulation.rst:48]: This doesn't appear as a link.

File: docs/source/API/core/numerics/bit-manipulation.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+.. |rotl| replace:: ``rotl``
+
+.. _rotr: https://en.cppreference.com/w/cpp/numeric/rotr
```
Comment (masterleinad) [inline docs/source/API/core/numerics/bit-manipulation.rst:52]: This doesn't appear as a link.

File: docs/source/API/core/numerics/bit-manipulation.rst (hunk 1 above)
Comment (dalg24) [reply docs/source/API/core/numerics/bit-manipulation.rst:24]: I know.  Don't think we can render them crossed out and had them in anticipation of adding later.  Want me to remove?

Comment (masterleinad) [reply docs/source/API/core/numerics/bit-manipulation.rst:24]: No, I think it's good to already have the links but maybe can find another way to mark unimplemented functions.

Comment (dalg24) [reply docs/source/API/core/numerics/bit-manipulation.rst:24]: > maybe can find another way to mark unimplemented functions

any suggestion?

File: docs/source/API/core/Numerics.rst [hunk 5]
```diff
@@ -24,3 +24,11 @@ library mathematical functions, such as ``fabs``, ``sqrt``, and ``sin``.
 
 The header ``<Kokkos_NumericTraits.hpp>`` implements a new facility that is being added to the C++23 standard library and
 is intended as a replacement for ``std::numeric_limits``.
+
+.. toctree::
+   :maxdepth: 1
+
+   numerics/bit-manipulation.md
```
Comment (ldh4) [inline docs/source/API/core/Numerics.rst:31]: Shouldn't this be `numerics/bit-manipulation.rst`?

Comment (dalg24) [reply docs/source/API/core/Numerics.rst:31]: That is a good point.  I copy/pasted from the same file without paying attention.  Want me to fix here or ok to deal with it as a follow up?

Comment (ldh4) [reply docs/source/API/core/Numerics.rst:31]: Ah looks like the links in `Numerics.rst` haven't been replaced along with the md to rst file conversions then. I think it's fine to fix those as a follow up.

**Comment by dalg24** [comment]
Ping

**Review by janciesko — APPROVED** [review]

**Pull request merged**